module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -3966,19 +3976,18 @@ zfs_fid(struct inode *ip, fid_t *fidp)
 	zfid_short_t	*zfid;
 	int		size, i, error;
 
-	ZFS_ENTER(zfsvfs);
+	if ((error = zfs_enter_verify_zp(zfsvfs, zp, FTAG)) != 0)
+		return (error);
 
 	if (fidp->fid_len < SHORT_FID_LEN) {
 		fidp->fid_len = SHORT_FID_LEN;
-		ZFS_EXIT(zfsvfs);
+		zfs_exit(zfsvfs, FTAG);
 		return (SET_ERROR(ENOSPC));
 	}
 
-	ZFS_VERIFY_ZP(zp);
```

Comment:
Let's preserve the existing behavior here and verify the zp after the fid length check.  It's possible the order is important, for more background see c074a7de1366656f77635da9db02cf5c09798cae

Reply:
done